csharp/src/Ice/IObjectPrx.cs

```diff
@@ -494,4 +525,308 @@ protected ObjectPrx(SerializationInfo info, StreamingContext context)
             IceReference = communicator.CreateReference(str, null);
         }
     }
+
+    // publicly visible Ice-internal class used as base class for request
+    // with non void return value
+    public class OutgoingRequestWithReturnValue<TOutParams>
```

Comment:
We should use the same name consistently for the operation parameters and return value, and so far we use "ParamList" and "ReturnValue", so the generic types could be `TParamList` and `TReturnValue`.

Reply:
fixed